packages/auth/test/integration/flows/firebaseserverapp.test.ts

```diff
@@ -0,0 +1,528 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import {
+  Auth,
+  createUserWithEmailAndPassword,
+  EmailAuthProvider,
+  getAdditionalUserInfo,
+  getRedirectResult,
+  GoogleAuthProvider,
+  onAuthStateChanged,
+  OperationType,
+  reauthenticateWithCredential,
+  signInAnonymously,
+  signInWithCredential,
+  signInWithCustomToken,
+  signInWithEmailAndPassword,
+  signInWithEmailLink,
+  signInWithRedirect,
+  signOut,
+  updateCurrentUser,
+  updateEmail,
+  updateProfile
+} from '@firebase/auth';
+import { isBrowser, FirebaseError } from '@firebase/util';
+import { initializeServerApp, deleteApp } from '@firebase/app';
+
+import {
+  cleanUpTestInstance,
+  getTestInstance,
+  getTestInstanceForServerApp,
+  randomEmail
+} from '../../helpers/integration/helpers';
+
+import { getAppConfig } from '../../helpers/integration/settings';
+
+use(chaiAsPromised);
+
+const signInWaitDuration = 200;
+
+describe('Integration test: Auth FirebaseServerApp tests', () => {
+  let auth: Auth;
+
+  beforeEach(() => {
+    auth = getTestInstance();
```

Comment:
Is this default instance ever used or are all the tests run on a getTestInstanceForServerApp() instance?

Reply:
Yes, the tests use this auth instance to create a new user (through varying means) and then uses that newly created user's idToken to initialize FirebaseServerApp.